File: eco-core/core-nms/v1_19_R1/src/main/java/com/willfp/ecoenchants/proxy/v1_19_R1/OpenInventory.java
package com.willfp.ecoenchants.proxy.v1_19_R1;

import com.willfp.ecoenchants.proxy.proxies.OpenInventoryProxy;
import org.bukkit.craftbukkit.v1_19_R1.entity.CraftPlayer;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public final class OpenInventory implements OpenInventoryProxy {
    @Override
    public Object getOpenInventory(@NotNull final Player player) {
        return ((CraftPlayer) player).getHandle().bU;
    }
}

File: eco-core/core-nms/v1_20_R2/src/main/kotlin/com/willfp/ecoenchants/proxy/v1_20_R2/OpenInventory.kt
package com.willfp.ecoenchants.proxy.v1_20_R2

import com.willfp.ecoenchants.mechanics.OpenInventoryProxy
import org.bukkit.craftbukkit.v1_20_R2.entity.CraftPlayer
import org.bukkit.entity.Player

class OpenInventory : OpenInventoryProxy {
    override fun getOpenInventory(player: Player): Any {
        return (player as CraftPlayer).handle.bS
    }
}
